src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java

```diff
@@ -130,6 +130,13 @@
  *     return field += "suffix"; // violation: Reference to instance variable "field" needs "this".
  *   }
  * }
+ *
+ * public static record C(int x) {
+ *     public C (int x){
+ *     // Overlapping by constructor argument.
+ *     x = x; // violation: Reference to instance variable "x" needs "this".
```

Comment:
1. indentation is wrong.

2. To be honest, such example with record and constructor seems confusing. May be change constructor to method?

Reply:
Fixed.